demos/config.h

```diff
@@ -29,16 +29,16 @@ typedef int MQTTNetworkContext_t;
 #define MQTT_MAX_QUEUED_PUBLISH_MESSAGES    10
 
 /* Include file for POSIX reference implementation. */
-#include "port/posix/iot_logging.h"
+#include "platform/include/iot_logging.h"
 
 /* Define the IotLog logging interface to enabling logging.
  * This demo maps the macro to the reference POSIX implementation for logging. */
-#define IotLog( messageLevel, pFormat, ... ) \
-    IotLog_Generic( messageLevel,            \
-                    "[%s:%d] [%s] "pFormat,  \
-                    __FILE__,                \
-                    __LINE__,                \
-                    LIBRARY_LOG_NAME,        \
+#define IotLog( messageLevel, pLibraryName, pFormat, ... ) \
```

Comment:
I do not think that we should add ```pLibraryName``` as a parameter to the ```IotLog``` macro. A customer can very well choose not to use ```pLibraryName```  in their logs by defining ```IotLog``` like:

```c
#define IotLog( messageLevel, pFormat, ... ) \
    IotLog_Generic( messageLevel,            \
                    "[%s:%d] "pFormat,  \
                    __FILE__,                \
                    __LINE__,                \
                    __VA_ARGS__ )
```
The customer will still be forced to define ```LIBRARY_LOG_NAME``` like:

```
#define LIBRARY_LOG_NAME "Customer App"
#include "iot_logging_setup.h"
```

On the other hand, if we kept it like before and removed the ```#ifndef LIBRARY_LOG_NAME``` check from iot_logging_setup.h, the customer who chooses not to include library names in their logs, will not be forced to define ```LIBRARY_LOG_NAME```.

Reply:
+1 to that. I think also this interface change will break all code that uses this function directly.  I found a bunch of places like this https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/2607cd8d609ef4fc61ff67a42eb1d6c2f7d420f2/libraries/standard/mqtt/src/iot_mqtt_serialize.c#L120

Reply:
The motivation behind adding the `pLibraryName` parameter to the interface was to remove exposing the `LIBRARY_LOG_NAME` macro in the `config.h` which may not carry any meaning to the customer, who looks at the config file without knowledge of the internal details of the logging macro framework. 
@cobusve, regarding the example shared on the breaking change, this PR is destined for the `development` branch where the new MQTT code, being re-written from scratch (along with the logging calls), is not expected to call the old `IotLog` interface that exists in the `v4_beta` branch.